src/client/Microsoft.Identity.Client/ApiConfig/Executors/ConfidentialClientExecutor.cs

```diff
@@ -71,14 +75,55 @@ public async Task<AuthenticationResult> ExecuteAsync(
             return await handler.RunAsync(cancellationToken).ConfigureAwait(false);
         }
 
+        private void ValidateAndConfigureMtlsPopForAcquireTokenForClient(
+            AcquireTokenForClientParameters clientParameters,
+            AcquireTokenCommonParameters commonParameters,
+            RequestContext requestContext)
+        {
+            if (clientParameters.UseMtlsPop)
+            {
+                // Throw if Authority != AAD for MTLS PoP as it is only supported for AAD
+                if (_confidentialClientApplication.AuthorityInfo.AuthorityType != AuthorityType.Aad)
+                {
+                    throw new MsalClientException(
+                        MsalError.InvalidAuthorityType,
+                        MsalErrorMessage.MtlsInvalidAuthorityTypeMessage);
+                }
+
+                // Validate that the certificate is not null for MTLS PoP
+                if (_confidentialClientApplication.Certificate == null)
+                {
+                    throw new MsalClientException(
+                        MsalError.MtlsCertificateNotProvided,
+                        MsalErrorMessage.MtlsCertificateNotProvidedMessage);
+                }
+
+                // Validate that a region is specified when MTLS PoP is enabled
+                bool isRegionMissing = string.IsNullOrEmpty(requestContext.ServiceBundle.Config.AzureRegion);
```

Comment:
nit: can skip creating this variable and put directly in the if condition. This variable is not used anywhere else.

Reply:
to me this is more readable as the variable name `isRegionMissing` gives a clear meaning/purpose to the condition,

Reply:
From perf perspective it doesn't matter, the compiler will optimize. 

From readability perspective... I tend to agree with Neha, this is too much info - you have a comment, a var name, and an exception. 

How about

```csharp
if (string.IsNullOrEmpty(requestContext.ServiceBundle.Config.AzureRegion))
{
    throw new MsalClientException(
                        MsalError.MtlsPopWithoutRegion,
                        "MTLS Proof of Possession requires a region to be specified. Use WithAzureRegion API to specifiy a region");
}
```